micrometer-test/src/test/java/io/micrometer/core/util/internal/logging/InternalMockLoggerFactoryTest.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2021 VMware, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.util.internal.logging;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Jonatan Ivanov
+ */
+class InternalMockLoggerFactoryTest {
+    @BeforeAll
+    static void setUpClass() {
+        InternalLoggerFactory.setDefaultFactory(new InternalMockLoggerFactory());
```

Comment:
Do we not need to clean up setting this global state? It seems like this could affect other tests.

Reply:
Instead of restoring the original one, I eliminated this completely from the tests by getting the logger directly from the `InternalMockLoggerFactory` through a convenience method. Also, I realized that this can cause issues during parallel test execution so I added support to inject the mock logger in a safe way.